### ISSUE TITLE
Fix QUERY MEMORY LIMIT is not honored during execution

### ISSUE THREAD

**Issue by jyenduri-uptycs** [memgraph/memgraph#977]
**Memgraph version**
2.8.0

**Environment**
Ubuntu 20.04.6 LTS (GNU/Linux 5.15.0-1031-aws x86_64)
memgraphdb running as docker

**Describe the bug**
Query memory limit is not honored. 

I have a complex query which does alot of filters, when we apply QUERY MEMORY LIMIT it not honored
```
MATCH (vpcCidr:AWSCidrBlock{tenantId: '1224445', customerId: '78787777'})<-[:BLOCK_ASSOCIATION]-(vpc:AWSVpc )<-[:PART_OF_VPC]-(n:EC2Instance {
  customerId: '78787777', tenantId: '1224445', exposedInternet: true
})-[:PART_OF_SUBNET]->(s:EC2Subnet)-[:PART_OF_ROUTE_TABLE]->(:RouteTable)<-[:PART_OF_ROUTE_TABLE]-(rtr:RouteTableRoute)-[:BLOCK_ASSOCIATION]->(target:AWSCidrBlock)
WHERE rtr.gatewayId='local' AND vpcCidr.cidrBlock = target.cidrBlock
WITH vpc, n
MATCH (vpc)<--(n2:EC2Instance{tenantId: '1224445', customerId: '78787777'})-[:PART_OF_SUBNET]->(:EC2Subnet)<-[:PART_OF_SUBNET]-(:AWSNetworkAcl)<-[:MEMBER_OF_NETWORK_ACL]-(nIbound:AWSNetworkAclInboundRule{action: 'allow'})<-[:MEMBER_OF_IP_NETWORK_INBOUND_RULE]-(naclBlock:AWSCidrBlock)
WITH vpc, n2, naclBlock, nIbound
MATCH (n2:EC2Instance)-[:MEMBER_OF_EC2_SECURITY_GROUP|NETWORK_INTERFACE*..2]->(group:EC2SecurityGroup)<-[:MEMBER_OF_EC2_SECURITY_GROUP]-(ipInbound:IpPermissionInbound)<-[:MEMBER_OF_IP_RULE]-(ipRange:IpRange)
WITH n2, naclBlock,group, nIbound, ipRange, ipInbound
MATCH (n:EC2Instance)-[:PART_OF_SUBNET]->(:EC2Subnet)<-[:PART_OF_SUBNET]-(:AWSNetworkAcl)<-[:MEMBER_OF_NETWORK_ACL]-(vulneroutbound:AWSNetworkAclEgressRule{ tenantId: '1224445', customerId: '78787777', action: 'allow'})
WITH n2, n, naclBlock, nIbound, ipRange, ipInbound, vulneroutbound
MATCH (n:EC2Instance {
  tenantId: '1224445', customerId: '78787777'
})-[:MEMBER_OF_EC2_SECURITY_GROUP|NETWORK_INTERFACE*..2]->(group:EC2SecurityGroup)<-[:MEMBER_OF_EC2_SECURITY_GROUP]-(vulnersgoutbound:IpPermissionEgress)
WITH n2, n, naclBlock, nIbound, ipRange, ipInbound,group, vulneroutbound, vulnersgoutbound
RETURN collect( DISTINCT n2) AS peers, n AS vulnerableInstance, vulneroutbound, vulnersgoutbound, naclBlock, nIbound AS naclInbound, ipRange, ipInbound AS sgInbound QUERY LIMIT 10 MB
```
We made sure we dont have any scan all in the EXPLAIN of query

**Expected behavior**
1. When query memory limit applied, one consume allocated memory 
2. If memory limit exceed query should fail and continue

**Comment by antoniofilipovic** [comment]
I think with the [new PR](https://github.com/memgraph/memgraph/pull/1250) query memory limit issue might be fixed. Did a quick test now and it worked. Will expand on this issue once I have more info.

**Comment by antoniofilipovic** [comment]
This PR should fix the issue. Memory limit was not being honored as we didn't properly track all allocations inside that query. https://github.com/memgraph/memgraph/pull/1340

**Comment by gitbuda** [comment]
TTL on this issue ~2 weeks (as soon as we release v2.12 it should be verified)

**Comment by as51340** [comment]
Closing since it is fixed with 2.12
